integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationTest.java

```diff
@@ -1549,4 +1554,412 @@ public void testUpdateTableWithReservedProperty() {
+      assertThat(currentLoadResponse.tableMetadata().schema().findField("recreated_timestamp")).isNotNull();
+
+      // Verify original schema fields are NOT present
+      assertThat(currentLoadResponse.tableMetadata().schema().findField("original_id")).isNull();
+      assertThat(currentLoadResponse.tableMetadata().schema().findField("original_name")).isNull();
+    }
+
+    // Test with current recreated ETag
+    Invocation currentEtagInvocation =
+        catalogApi
+            .request("v1/" + currentCatalogName + "/namespaces/ns1/tables/test_drop_recreate_behavior_table")
+            .header(HttpHeaders.IF_NONE_MATCH, recreatedETag)
+            .build("GET");
+
+    try (Response currentEtagResponse = currentEtagInvocation.invoke()) {
+      // Should return 304 Not Modified because ETag matches current recreated table
+      assertThat(currentEtagResponse.getStatus()).isEqualTo(Response.Status.NOT_MODIFIED.getStatusCode());
+      assertThat(currentEtagResponse.hasEntity()).isFalse();
+    }
+  }
+
+  @Test
+  public void testETagChangeAfterDMLOperations() {
+    // Test that ETags change after DML operations (INSERT, UPDATE, DELETE)
+    Namespace ns1 = Namespace.of("ns1");
+    restCatalog.createNamespace(ns1);
+    TableIdentifier tableId = TableIdentifier.of(ns1, "test_dml_etag_table");
+
+    // Create table with initial schema
+    Schema schema = new Schema(List.of(
+        Types.NestedField.required(1, "id", Types.LongType.get()),
+        Types.NestedField.optional(2, "name", Types.StringType.get()),
+        Types.NestedField.optional(3, "value", Types.IntegerType.get())
+    ));
+    restCatalog.buildTable(tableId, schema).create();
+
+    // Load table and get initial ETag (before any data)
+    Invocation initialInvocation =
+        catalogApi
+            .request("v1/" + currentCatalogName + "/namespaces/ns1/tables/test_dml_etag_table")
+            .build("GET");
+
+    String initialETag;
+    String initialMetadataLocation;
+    try (Response initialResponse = initialInvocation.invoke()) {
+      assertThat(initialResponse.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
+      assertThat(initialResponse.getHeaders()).containsKey(HttpHeaders.ETAG);
+      initialETag = initialResponse.getHeaders().getFirst(HttpHeaders.ETAG).toString();
+
+      LoadTableResponse initialLoadResponse = initialResponse.readEntity(LoadTableResponse.class);
+      initialMetadataLocation = initialLoadResponse.metadataLocation();
+    }
+
+    // Simulate DML operation by creating a new snapshot (append operation)
+    Table table = restCatalog.loadTable(tableId);
+
+    // Create a data file and append it (simulating INSERT operation)
+    AppendFiles append = table.newAppend();
+
+    // Create a mock data file entry
+    DataFile dataFile = DataFiles.builder(table.spec())
+        .withPath("/path/to/data/file1.parquet")
```

Comment:
not sure this will work with cloud storage :thinking: Why not `table.locationProvider().newDataLocation(...)`?

Reply:
True ! Fixed it.